### DISCUSSION TITLE
Pi4J 2.5 multiple I/O providers

### DISCUSSION THREAD

**Discussion by tatery** [Pi4J/pi4j#359] [Q&A]
Hi,
Providers description says:
"_As of Pi4J 2.5 multiple providers for the same I/O type is no longer supported. During the Context initialization it will ensure only a single provider for an I/O type is loaded._" (https://pi4j.com/documentation/providers/)

I'm using the built-in RaspberryPi I/O (with PiGpio provider) and the I/O extension MCP23017 (along with a custom GPIO provider: https://github.com/savageautomate/pi4j-plugin-microchip).

Is it possible to use both providers at the same time in Pi4J version 2.5?

**Comment by taartspi** [comment]
@eitch   Can you read over this question?  With provider priority implementation it seems this provider must be a new IO type.  Your thoughts

**Comment by eitch** [comment]
At the moment it isn't possible. This is a use-case i didn't see. We'll have to think about this. Currently we've done this differently, e.g. @taartspi has a MCP23017 implementation here: 
* https://github.com/Pi4J/pi4j-example-devices/blob/master/src/main/java/com/pi4j/devices/mcp23017/README.md
* https://github.com/Pi4J/pi4j-example-devices/blob/master/src/main/java/com/pi4j/devices/mcp23017/Mcp23017App.java
